File: nom-api/Nom.Orch/Models/MealPlan/MealPlanRuleResponseModel.cs
// File: Nom.Orch/Models/MealPlan/MealPlanRuleResponseModel.cs

namespace Nom.Orch.Models.MealPlan
{
    public class MealPlanRuleResponseModel
    {
        public long Id { get; set; }
        public long HouseholdId { get; set; }
        public long DayOfWeekId { get; set; }
        public string DayOfWeek { get; set; } = string.Empty;
        public long MealTypeId { get; set; }
        public string MealType { get; set; } = string.Empty;
        public string QueryFilterString { get; set; } = string.Empty;
        public DateTime CreatedDate { get; set; }
        public DateTime? ModifiedDate { get; set; }
    }
} 

File: nom-ui/src/app/meal-plan/models/meal-plan-rule-response.model.ts
// File: nom-ui/src/app/meal-plan/models/meal-plan-rule-response.class.ts

import { IMealPlanRuleResponseModel } from './meal-plan-rule-response.model.interface';

export class MealPlanRuleResponseModel implements IMealPlanRuleResponseModel {
    id = 0;
    name = '';
    description?: string;
    dayOfWeekId = 0;
    dayOfWeek?: string;
    dayOfWeekName?: string;
    mealTypeId = 0;
    mealType?: string;
    mealTypeName?: string;
    queryFilterString?: string;
    householdId = 0;
    isActive = true;
    createdDate: Date = new Date();
    modifiedDate?: Date;

    constructor(data?: Partial<IMealPlanRuleResponseModel>) {
        if (data) {
            Object.assign(this, data);
        }
    }
} 